src/status_im/ui/screens/browser/events.cljs

```diff
@@ -0,0 +1,76 @@
+(ns status-im.ui.screens.browser.events
+  (:require status-im.ui.screens.browser.navigation
+            [status-im.utils.handlers :as handlers]
+            [status-im.data-store.browser :as browser-store]
+            [re-frame.core :as re-frame]
+            [status-im.utils.random :as random]
+            [status-im.i18n :as i18n]))
+
+(re-frame/reg-cofx
+  :all-stored-browsers
+  (fn [cofx _]
+    (assoc cofx :all-stored-browsers (browser-store/get-all))))
+
+(handlers/register-handler-fx
+  :initialize-browsers
+  [(re-frame/inject-cofx :all-stored-browsers)]
+  (fn [{:keys [db all-stored-browsers]} _]
+    (let [{:accounts/keys [account-creation?]} db]
+      (when-not account-creation?
+        (let [browsers (into {} (map #(vector (:browser-id %) %) all-stored-browsers))]
+          {:db (assoc db :browser/browsers browsers)})))))
+
+(re-frame/reg-fx
+  :save-browser
+  (fn [browser]
+    (browser-store/save browser)))
+
+(defn get-new-browser [browser now]
+  (cond-> browser
+          true
+          (assoc :timestamp now)
+          (not (:browser-id browser))
+          (assoc :browser-id (random/id))
+          (not (:name browser))
+          (assoc :name (i18n/label :t/browser))))
+
+(handlers/register-handler-fx
+  :add-browser
```

Comment:
This looks like event which is only ever dispatched as part of other events (`:open-dapp-in-browser`, `:open-browser` and `:update-browser`) - looks like it can be coded as function `add-browser` instead and those 3 events will just call it and merge the effects map produced by the function to their own effects.

Reply:
yes ,thanks